src/run_tribler.py

```diff
@@ -60,38 +74,52 @@ async def main() -> None:
 
     root_state_dir = get_root_state_directory(os.environ.get('TSTATEDIR', 'state_directory'))
     logger.info("Root state dir: %s", root_state_dir)
-
-    api_port, api_key = int(os.environ.get('CORE_API_PORT', '0')), os.environ.get('CORE_API_KEY')
-
     config = TriblerConfigManager(root_state_dir / "configuration.json")
     config.set("state_dir", str(root_state_dir))
 
-    if config.get("api/refresh_port_on_start"):
-        config.set("api/http_port", 0)
-        config.set("api/https_port", 0)
-
-    if api_key is None and config.get("api/key") is None:
-        api_key = os.urandom(16).hex()
-
-    if api_key is not None and api_key != config.get("api/key"):
-        config.set("api/key", api_key)
+    if "CORE_API_PORT" in os.environ:
+        config.set("api/http_port", int(os.environ.get("CORE_API_PORT")))
         config.write()
 
-    if api_port is not None and api_port != config.get("api/http_port"):
-        config.set("api/http_port", api_port)
+    if "CORE_API_KEY" in os.environ:
+        config.set("api/key", os.environ.get("CORE_API_KEY"))
         config.write()
 
-    logger.info("Start tribler core. API port: %d. API key: %s.", api_port, config.get("api/key"))
+    if config.get("api/key") is None:
+        config.set("api/key", os.urandom(16).hex())
+        config.write()
 
+    logger.info("Creating session. API port: %d. API key: %s.", config.get("api/http_port"), config.get("api/key"))
     session = Session(config)
+
+    torrent_uri = parsed_args.get('torrent')
+    if torrent_uri and os.path.exists(torrent_uri) and torrent_uri.endswith(".torrent"):
+        torrent_uri = Path(torrent_uri).as_uri()
```

Comment:
When looking at this, I just recalled that the Windows installer also [registers itself for the magnet extension](https://github.com/Tribler/tribler/blob/42a8ac190f8f9f8bc4733c34c8092dd1375dd746/build/win/resources/tribler.nsi#L184-L190). Do you want to add support for that in this PR or should be open a separate issue?

Reply:
Another alternative, "none of the above", is to remove support for `.magnet` files.

Reply:
`DownloadsEndpoint.add_download` calls `DownloadManager.start_download_from_uri` which seems to have the ability to deal with magnet links. So, I'm hoping this will be fixed too. I haven't tested it, though.

Reply:
I guess we/you can add `torrent_uri.endswith(".magnet")` to this filter then and if anything breaks we'll just open an issue.

Reply:
Done, .magnet support should work now. I also fixed an issue with the default hop count.